include/control_toolbox/pid_ros_impl.hpp

```diff
@@ -31,32 +31,43 @@
  *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
-#ifndef CONTROL_TOOLBOX__PIDROS_IMPL_HPP_
-#define CONTROL_TOOLBOX__PIDROS_IMPL_HPP_
+#ifndef CONTROL_TOOLBOX__PID_ROS_IMPL_HPP_
+#define CONTROL_TOOLBOX__PID_ROS_IMPL_HPP_
 
 #include <algorithm>
 #include <cmath>
+#include <memory>
 #include <string>
 #include <utility>
 #include <vector>
 
 #include "control_toolbox/pid_ros.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp_lifecycle/lifecycle_publisher.hpp"
 
 namespace control_toolbox
 {
 
 template<typename NodeT>
 PidROS<NodeT>::PidROS(std::shared_ptr<NodeT> node_ptr, std::string topic_prefix)
-:node_(node_ptr)
+: node_(node_ptr)
 {
   if (topic_prefix.back() != '.' && !topic_prefix.empty()) {
     topic_prefix_ = topic_prefix + ".";
   } else {
     topic_prefix_ = topic_prefix;
   }
 
-  state_pub_ =  node_->create_publisher<control_msgs::msg::PidState>(
+  state_pub_ = node_->template create_publisher<control_msgs::msg::PidState>(
     topic_prefix + "/pid_state", rclcpp::SensorDataQoS());
+  rclcpp_lifecycle::LifecycleNode::SharedPtr is_lifecycle_node =
+    std::dynamic_pointer_cast<rclcpp_lifecycle::LifecycleNode>(node_);
+  if (is_lifecycle_node) {
+    auto state_pub_lifecycle_ =
+      std::dynamic_pointer_cast<rclcpp_lifecycle::LifecyclePublisher<control_msgs::msg::PidState>>(
+      state_pub_);
+    state_pub_lifecycle_->on_activate();
+  }
```

Comment:
This shouldn't be handled within the PID controller.
The lifecycle activation/deactivation should be done outside of this class.